pkg/virt-controller/watch/vm_test.go

```diff
@@ -5353,6 +5367,354 @@ var _ = Describe("VirtualMachine", func() {
 				})
 			})
 		})
+
+		Context("The RestartRequired condition", func() {
+			var vm *virtv1.VirtualMachine
+			var vmi *virtv1.VirtualMachineInstance
+			var kv *virtv1.KubeVirt
+			var crList appsv1.ControllerRevisionList
+			var crListLock sync.Mutex
+
+			expectRestartRequired := func(shouldExpect bool) {
+				Eventually(func() {
+					for _, obj := range vmInformer.GetStore().List() {
+						vmObj := obj.(*virtv1.VirtualMachine)
+						if vmObj.ObjectMeta.UID == vm.ObjectMeta.UID {
+							found := false
+							for _, cond := range vmObj.Status.Conditions {
+								if cond.Type == virtv1.VirtualMachineRestartRequired {
+									found = true
+									break
+								}
+							}
+							Expect(found).To(Equal(shouldExpect), "unexpected state of the RestartRequired condition")
+						}
+					}
+				})
```

Comment:
```suggestion
				Eventually(func() {
					obj, exists, _ := vmInformer.GetStore().GetByKey(vm.Namespace + "/" + vm.Name)
					Expect(exists).To(BeTrue())
					found := false
					vmObj := obj.(*virtv1.VirtualMachine)
					for _, cond := range vmObj.Status.Conditions {
						if cond.Type == virtv1.VirtualMachineRestartRequired {
							found = true
							break
						}
					}
					Expect(found).To(Equal(shouldExpect), "unexpected state of the RestartRequired condition")
				})
```

Reply:
That doesn't seem to work :sweat_smile:

Reply:
Strange :thinking: I was pretty sure it was working!

Reply:
I blindly applied the diff and ran `WHAT=pkg/virt-controller/watch make go-test`, got 2 test failures.
The informer might not be mocked well enough for your version...